### DISCUSSION TITLE
Logs stop at irregular intervals, running high state allows some through

### DISCUSSION THREAD

**Discussion by thunderwood19** [Security-Onion-Solutions/securityonion#3485] [Unsupported Versions] [answered]
As the title suggests, I am having an issue keeping logs coming in.  Each morning, I run a "salt \* state.apply" then so-elastic-restart on the manager which gets some through, but then stops again. so-elastic-diagnose attached, w/ global.sls and manager.sls

[elastic_log.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6156363/elastic_log.txt)

host:
  mainint: 'ens192'
sensoroni:
  node_address: '10.1.2.101'
  node_description: ''

patch:
  os:
    schedule_name: 'manual'
    enabled: True
    splay: 300

manager:
  mainip: '10.1.2.101'
  mainint: 'ens192'
  esheap: '30000m'
  esclustername: '{{ grains.host }}'
  freq: 0
  domainstats: 0
  mtu:
  elastalert: 1
  es_port: 9200
  log_size_limit: 7000
  cur_close_days: 30
  grafana: 1
  osquery: 1
  thehive: 1
  playbook: 1

elasticsearch:
  mainip: '10.1.2.101'
  mainint: 'ens192'
  esheap: '25000m'
  esclustername: '{{ grains.host }}'
  node_type: ''
  es_port: 9200
  log_size_limit: 7000
  node_route_type: 'hot'
logstash_settings:
  ls_pipeline_batch_size: 125
  ls_input_threads: 4
  lsheap: 6000m
  ls_pipeline_workers: 15

idstools:
  config:
    ruleset: 'ETOPEN'
    oinkcode: ''
    urls:
  sids:
    enabled:
    disabled:
    modify:

kratos:
  kratoskey: '********************88'

  global:
  soversion: 2.3.30
  hnmanager: '10.0.0.0/8,192.168.0.0/16,172.16.0.0/12'
  ntpserver: '10.1.1.113'
  dockernet: '172.17.0.0'
  proxy: ''
  mdengine: 'ZEEK'
  ids: 'Suricata'
  url_base: '01p-secon04'
  managerip: '10.1.2.101'
  airgap: False
  hivekey: *
  hiveplaysecret: *
  cortexkey: *
  cortexorgname: 'SecurityOnion'
  cortexorguserkey: *
  cortexplaysecret:*
  fleet_custom_hostname:
  fleet_manager: True
  fleet_node: False
  fleet_packages-timestamp: '2021-03-11-17:42'
  fleet_packages-version: 3
  fleet_hostname: '01p-secon04'
  fleet_ip: '10.1.2.101'
  sensoronikey: *
  wazuh: 1
  managerupdate: 1
  imagerepo: 'security-onion-solutions'
  pipeline: 'redis'
sensoroni:
  node_checkin_interval_ms: 1000
strelka:
  repos:
    - https://github.com/Neo23x0/signature-base
  enabled: 1
  rules: 1
curator:
  hot_warm: false
elastic:
  features: True
elasticsearch:
  replicas: 0
  true_cluster: false
  true_cluster_name: 'so'
  discovery_nodes: 1
  hot_warm_enabled: false
  cluster_routing_allocation_disk.threshold_enabled: true
  cluster_routing_allocation_disk_watermark_low: '95%'
  cluster_routing_allocation_disk_watermark_high: '98%'
  cluster_routing_allocation_disk_watermark_flood_stage: '98%'
  index_settings:
    so-beats:
      shards: 3
      warm: 3
      close: 15
      delete: 30
    so-firewall:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-flow:
      shards: 1
      warm: 3
      close: 15
      delete: 365
    so-ids:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-import:
      shards: 1
      warm: 7
      close: 73000
    so-ossec:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-strelka:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-syslog:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-zeek:
      shards: 5
      warm: 7
      close: 365
      delete: 45
minio:
  access_key: ''
  access_secret: ''
s3_settings:
  size_file: 2048
  time_file: 1
  upload_queue_size: 4
  encoding: 'gzip'
  interval: 5
backup:
  locations:
    - /opt/so/saltstack/local
soctopus:
  playbook:
    rulesets:
      - windows
      - application
      - generic
      - network
      - web
docker:
  range: '172.17.0.0/24'
  bip: '172.17.0.1/24'
redis_settings:
  redis_maxmemory: 2000
playbook:
  api_key: *

**Comment by TOoSmOotH** [comment, marked as answer]
Looks like you need to increase your heap size. 

1 WARN ][io.netty.channel.DefaultChannelPipeline] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception. java.lang.OutOfMemoryError: Direct buffer memory [2021-03-12T18:24:51,792][INFO ][org.logstash.beats.BeatsHandler] [local: 0.0.0.0:5044, remote: undefined] Handling exception: java.lang.OutOfMemoryError: Direct buffer memory (caused by: java.lang.OutOfMemoryError: Direct buffer memory) [2021-03-12T18:24:53,652][INFO ][org.logstash.beats.BeatsHandler] [local: 0.0.0.0:5044, remote: undefined] Handling exception: java.lang.OutOfMemoryError: Direct buffer memory (caused by: java.lang.OutOfMemoryError: Direct buffer memory) [2021-03-12T18:24:53,719][INFO ][org.logstash.beats.BeatsHandler] [local: 0.0.0.0:5644, remote: 10.1.2.105:34746] Handling exception: java.lang.OutOfMemoryError: Direct buffer memory (caused by: java.lang.OutOfMemoryError: Direct buffer memory) [2021-03-12T18:24:53,653][

**Reply by thunderwood19** [reply]
New it was something silly, Thankyou sir!
